File: pdlearning/src/microservices/uploader-microservice/Api/Application/Models/PersonalSpaceModel.cs
using System;
using Microservice.Uploader.Domain.Entities;
using Thunder.Platform.Core.Domain.Auditing;

namespace Microservice.Uploader.Application.Models
{
    public class PersonalSpaceModel
    {
        public PersonalSpaceModel()
        {
        }

        public PersonalSpaceModel(PersonalSpace entity)
        {
            this.Id = entity.Id;
            this.UserId = entity.UserId;
            this.TotalSpace = entity.TotalSpace;
            this.TotalUsed = entity.TotalUsed;
            this.IsStorageUnlimited = entity.IsStorageUnlimited;
        }

        public Guid Id { get; set; }

        public Guid UserId { get; set; }

        public double TotalSpace { get; set; }

        public double TotalUsed { get; set; }

        public bool IsStorageUnlimited { get; set; }
    }
}

File: pdlearning/src/web-apps/src/libraries/domain-api/src/personal-space/models/personal-space.model.ts
export interface IPersonalSpaceModel {
  userId: string;
  totalSpace: number;
  totalUsed: number;
  isStorageUnlimited: boolean;
}

export class PersonalSpaceModel implements IPersonalSpaceModel {
  public userId: string;
  public totalSpace: number;
  public totalUsed: number;
  public isStorageUnlimited: boolean;

  constructor(data?: IPersonalSpaceModel) {
    if (data != null) {
      this.userId = data.userId;
      this.totalSpace = data.totalSpace;
      this.totalUsed = data.totalUsed ? data.totalUsed : 0;
      this.isStorageUnlimited = data.isStorageUnlimited;
    }
  }
}
